R/systemfit-tidiers.R

```diff
@@ -0,0 +1,52 @@
+#' @templateVar class systemfit
+#' @template title_desc_tidy
+#' 
+#' @param x A `systemfit` object produced by a call to [systemfit::systemfit()].
+#' @param conf.int Logical indicating whether or not to include a confidence interval in the tidied output. Defaults to `FALSE`.
+#' @template param_unused_dots
+#' 
+#' @evalRd return_tidy(
+#'   "term", 
+#'   "estimate", 
+#'   "std.error", 
+#'   "p.value",
+#'   "conf.low",
+#'   "conf.high"
+#' )
+#'
+#' @details This tidy method works with any model objects of class `systemfit`. 
+#'          Default returns a tibble of six columns.
+#' @importFrom stats confint
+#' @examples
+#' library(systemfit)
+#' df <- data.frame(X = rnorm(100), Y = rnorm(100), Z = rnorm(100), W = rnorm(100))
+#'
+#` fit <- systemfit(formula = list(Y ~ Z, W ~ X), data = df, method = "SUR")
+#'
+#' tidy(fit)
+#' tidy(fit, conf.int = TRUE)
+#' @export
+#' @seealso [tidy()], [systemfit::systemfit()]
+#' @family systemfit tidiers
+#' @aliases systemfit_tidiers
+tidy.systemfit <- function(x, conf.int = TRUE, ...) {
+  nn <- c("term", "estimate", "std.error", "p.value")
+  
+  sf_summary <- summary(x)
+  sf_coefs <- as.data.frame(sf_summary$coefficients)
+  sf_coefs <- sf_coefs[,- which(names(sf_coefs) == "t value")]
```

Comment:
Is there a reason to strip out the T statistic? If not, let's keep it and call that column `statistic`.

Reply:
Done in 31b55e4b2bc776f2f36b6fa3d2c3959c0c624824